### DISCUSSION TITLE
Some problems in the scheduling of microtask updates and react updates

### DISCUSSION THREAD

**Discussion by lsbFlying** [pmndrs/valtio#696] [General]
I read the source code and learned that the way our valtio finally implements batch updates is combined with micro-task processing, which coincides with the way I guessed when I looked at the source code at first 😄, but it seems to cause input problems. Examples are as follows:

```js
import {proxy, useSnapshot} from "./valtio";

const s = proxy({
  text: "hello",
});

function V() {
  const { text } = useSnapshot(s);
  return (
    <input
      value={text}
      onChange={(event) => {
        s.text = event.target.value;
      }}
    />
  );
}
```

As a result, we seem to be able to enter only English in the input language, but we can't enter a language that is not unexpected in English. I don't know if the author has tried, or it's just a problem caused by my computer.

I'm not sure if it's deliberately designed this way, or has it discussed this problem and there are other solutions.

**Comment by dai-shi** [comment]
https://github.com/pmndrs/valtio/blob/7fe26c2a8ff2df8a5b8df019f8878dbf48e40876/readme.md?plain=1#L221-L234
